src/utils/slot.py

```diff
@@ -88,7 +91,7 @@ def get_first_non_missed_slot(
     if not existed_header:
         raise NoSlotsAvailable('No slots available for current report. Check your CL node.')
 
-    if ref_slot_is_missed:
+    if ref_slot_is_missed and direction == 'back':
```

Comment:
we need sanity check for `forward` mode as well, that the slot by `parent_root` of the found one is less or equal `slot`

Reply:
Added.